File: frontend/src/types/projects.ts
export interface ProjectConfig {
  id: string;
  externalProjectId: string;
  name: string;
  externalWorkspaceId: string;
  enabled: boolean;
  pickupColumn: string;
  reviewColumn: string;
  doneColumn: string;
  progressColumn: string;
  promptTemplate?: string | null;
  repoFullNames?: string[];
  repoUrls?: string[];
  agentsMd?: string | null;
  reviewEnabled?: boolean | null;
  reviewMaxTurns?: number | null;
  reviewPromptTemplate?: string | null;
  maxInProgressTasks?: number | null;
  createdAt: string;
  providerId?: string;
}

export interface CreateProjectRequest {
  externalProjectId: string;
  externalWorkspaceId?: string;
  name?: string;
  providerId: string;
  enabled?: boolean;
  pickupColumn?: string;
  progressColumn?: string;
  reviewColumn?: string;
  doneColumn?: string;
  promptTemplate?: string | null;
  repoFullNames?: string[];
  agentsMd?: string | null;
  reviewEnabled?: boolean | null;
  reviewMaxTurns?: number | null;
  reviewPromptTemplate?: string | null;
  maxInProgressTasks?: number | null;
}

export interface UpdateProjectRequest {
  enabled?: boolean;
  externalWorkspaceId?: string;
  name?: string;
  pickupColumn?: string;
  progressColumn?: string;
  reviewColumn?: string;
  doneColumn?: string;
  promptTemplate?: string | null;
  repoFullNames?: string[];
  agentsMd?: string | null;
  reviewEnabled?: boolean | null;
  reviewMaxTurns?: number | null;
  reviewPromptTemplate?: string | null;
  maxInProgressTasks?: number | null;
  providerId?: string;
}

export interface ColumnInfo {
  id: string;
  name: string;
  slug: string;
}

export interface ColumnsResponse {
  columns: ColumnInfo[];
}

export interface IntegrationProvider {
  id: string;
  name: string;
  providerType: string;
  instanceUrl: string;
  apiKey: string;
  createdAt: string;
}

export interface CreateProviderRequest {
  name: string;
  providerType?: string;
  instanceUrl: string;
  apiKey: string;
}

export interface UpdateProviderRequest {
  name?: string;
  providerType?: string;
  instanceUrl?: string;
  apiKey?: string;
}

export interface LookupProjectResponse {
  id: string;
  name: string;
  slug: string;
  columns: ColumnInfo[];
}

export interface WorkspaceInfo {
  id: string;
  name: string;
}

export interface ProjectInfo {
  id: string;
  name: string;
  slug: string;
}

File: shared/src/api/app/projects.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Clone, Debug, Eq, PartialEq, Deserialize)]
pub struct AppProject {
    pub id: Uuid,
    pub external_project_id: String,
    pub name: String,
    pub external_workspace_id: String,
    pub enabled: bool,
    #[serde(default)]
    pub pickup_column: String,
    #[serde(default)]
    pub progress_column: String,
    #[serde(default)]
    pub review_column: String,
    #[serde(default)]
    pub done_column: String,
    pub repo_full_names: Vec<String>,
    pub provider_id: Option<Uuid>,
}

#[derive(Clone, Debug, Eq, PartialEq, Deserialize)]
pub struct ProviderWorkspace {
    pub id: String,
    pub name: String,
}

#[derive(Clone, Debug, Eq, PartialEq, Deserialize)]
pub struct ProviderProject {
    pub id: String,
    pub name: String,
    pub slug: String,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize)]
pub struct CreateProjectRequest {
    pub external_project_id: String,
    pub external_workspace_id: String,
    pub name: String,
    pub provider_id: Uuid,
    pub enabled: bool,
    pub repo_full_names: Vec<String>,
}

#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize)]
pub struct UpdateProjectRequest {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub enabled: Option<bool>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub pickup_column: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub progress_column: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub review_column: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub done_column: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub repo_full_names: Option<Vec<String>>,
}
